### DISCUSSION TITLE
How to limit the highest quality when setting it to "auto" ?

### DISCUSSION THREAD

**Discussion by chethavuthy** [vidstack/player#1240] [Q&A]
for example, i have a playlist of 360p, 480p, 720p, 1080p. when setting the quality to auto, how to conditionally limit the highest quality to only 480p?

**Comment by mihar-22** [comment]
The best way to do it whilst supporting all devices is to build your own playlist client-side or using service workers. I actually just [Tweeted](https://twitter.com/matt_mcclure/status/1777613337146057080) about it, and Matt linked to an article in which he explores solutions with service workers.

**Reply by chethavuthy** [reply]
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-STREAM-INF:BANDWIDTH=427549,RESOLUTION=426x240,CODECS="avc1.4d4015,mp4a.40.2"
240p.m3u8

#EXT-X-STREAM-INF:BANDWIDTH=870782,RESOLUTION=640x360,CODECS="avc1.4d401e,mp4a.40.2"
360p.m3u8

#EXT-X-STREAM-INF:BANDWIDTH=1463798,RESOLUTION=852x480,CODECS="avc1.4d401f,mp4a.40.2"
480p.m3u8
```

got the `Event Type: hlsError`, is my playlist incorrect?
it hits the network to `blob:http://localhost:8000/360p.m3u8` while the URL is `blob:http://localhost:8000/81f6d60e-a086-4b4c-9e51-f3fa75ceab16`

**Reply by mihar-22** [reply]
Yes, you need absolute URL to the rendition manifests. Replace 240p and so on with https://…
